### ISSUE TITLE
[Feature Request]: Add dispersion to ASE calculator

### ISSUE THREAD

**Issue by ajhoffman1229** [CederGroupHub/chgnet#188]
### Email (Optional)

ajhoff29@mit.edu

### Problem

I would like to be able to add a dispersion/van der Waals correction to the pre-trained CHGNet model. A similar option is available in the MACE-MP-0 model [here](https://github.com/ACEsuit/mace/blob/575af0171369e2c19e04b115140b9901f83eb00c/mace/calculators/foundations_models.py#L112).

### Proposed Solution

I believe that an option can be added to the CHGNetCalculator for a dispersion correction similar to the option in [MACE-MP-0](https://github.com/ACEsuit/mace/blob/575af0171369e2c19e04b115140b9901f83eb00c/mace/calculators/foundations_models.py#L112). Their implementation requires installing [torch-dftd](https://github.com/pfnet-research/torch-dftd), and creates an instance of an ASE `SumCalculator` that adds the energies, forces, and stresses of each contribution together.

### Alternatives

If this feature cannot be added, it would be great if a note could be added to the README pointing users to the torch-dftd implementation in the event that they want to add a dispersion correction to CHGNet independent of the repo. Currently, the python packages for DFT-D3 and [DFT-D4](https://github.com/dftd4/dftd4) dispersion models from the Grimme group are not compatible with torch, so the conventional method of adding the dispersion does not work with CHGNet.

Here is some example code where this works with CHGNet and the torch-dftd implementation:
```python
from ase.build import fcc111
from ase.calculators.mixing import SumCalculator
from chgnet.model.dynamics import CHGNetCalculator
from chgnet.model.model import CHGNet
from torch_dftd.torch_dftd3_calculator import TorchDFTD3Calculator

# pre-trained chgnet model
chgpotential = CHGNet.load()
chgcalc = CHGNetCalculator(model=chgpotential)

# Calculate reference values:
atoms = fcc111("Cu", (2, 2, 1), vacuum=10.0)
atoms[0].x += 0.2

chgd3 = SumCalculator([chgcalc, TorchDFTD3Calculator()])
atoms.calc = chgd3
stress = atoms.get_stress()
print(chgd3.name)
print(atoms.get_chemical_formula(), stress.shape, stress)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by BowenD-UCB** [comment]
@ajhoffman1229 Can you please give some tests the DFTD and SumCalculator and share a notebook example or open a PR? Happy to help and merge if the combined calculator works stably on your side

**Comment by ajhoffman1229** [comment]
Yes, I can make a notebook and submit a PR so it can be added as an example to the repo. The code above worked when I tested it locally. I will try to polish it later this week and submit for review.

**Comment by ajhoffman1229** [comment]
Hi folks, I am not sure what I did but the code that I wrote earlier this week now seems to be working now and I can compute energies with a CHGNet+D4 `SumCalculator` instance in ASE, in addition to the torch-dftd D3 calculator. There does not appear to be the incompatibility I was running into earlier, and I apologize for any wild goose chases opening this issue might have caused. I made a notebook tutorial showing how to add a dispersion correction, which is in #192.
